### PR TITLE
add validation for k8s agentpool dnsprefix

### PR DESCRIPTION
**Pull request by rjtsdl**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/534)

### CODE REVIEW

**Comment by rjtsdl** [comment]
Here, i only do validation fix. The compatible change need to be made somewhere else.

File: pkg/api/v20160930/validate.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ func (a *AgentPoolProfile) Validate() error {
 	if e := validateName(a.VMSize, "AgentPoolProfile.VMSize"); e != nil {
 		return e
 	}
+	if orchestratorType == Kubernetes {
```
Comment (JiangtianLi) [inline pkg/api/v20160930/validate.go]: We need to do case insensitive comparison. See #529. You can check in first and I'll change.

**Comment by weinong** [comment]
why is it different than vlab validation?

**Comment by rjtsdl** [comment]
vlab use len(ports) to determine whether the DNSPrefix could be empty or not.
In version 2017-01-31 and 2016-09-30, ports is not in the api model. We have to do things in another way. We use DNSPrefix to determine ports should be set or not. 

However, I don't see why we cannot keep the principle consistent in vlab. I can also do validate change in vlab, let it also use DNSPrefix to determine ports.

@amanohar , how do you think?

File: pkg/api/v20160930/validate.go (hunk 1 above)
Comment (rjtsdl) [inline pkg/api/v20160930/validate.go]: I approved your pr, if you get checked in earlier. I will modify upon it.

Comment (JiangtianLi) [inline pkg/api/v20160930/validate.go]: I am working on a solution that doesn't need to change string comparison. So feel free to check in.

**Comment by amanohar** [comment]
@rjtsdl, yes I think validation should be consistent through all the API versions. Let's discuss tomorrow (and I agree with what you mentioned about validation being the same throughout).

**Comment by rjtsdl** [comment]
@acs-bot test this please

**Comment by rjtsdl** [comment]
@amanohar @weinong , I intentionally make all the version's DNSPrefix validation follow the same philosophy. They have subtle difference, mainly because of the version's difference.

**Comment by rjtsdl** [comment]
@acs-bot test this please

File: pkg/api/vlabs/validate.go [hunk 2]
```diff
@@ -290,6 +303,13 @@ func (a *Properties) validateNetworkPolicy() error {
 	return nil
 }
 
+func validateNameEmpty(name string, label string) error {
```
Comment (amanohar) [inline pkg/api/vlabs/validate.go]: validateFieldEmpty or validateStringEmpty is better IMO.

Comment (rjtsdl) [inline pkg/api/vlabs/validate.go]: Saw this, after i merged it. We can do another one to correct the name. There is validateName as well. It needs a broader change.

**Pull request merged**